props/saber_BC_buttons.h

```diff
@@ -386,36 +1161,248 @@ EFFECT(volmax);     // for maximum volume reached
 #warning The faston.wav sound will be replaced with fastout.wav. If you have a good reason to keep faston.wav as is, please post at https://crucible.hubbe.net/
 EFFECT(faston);     // for EFFECT_FAST_ON. Being replaced by fastout.wav, which is already defined in the main OS.
 #endif
-EFFECT(blstbgn);    // for Begin Multi-Blast
-EFFECT(blstend);    // for End Multi-Blast
 EFFECT(push);       // for Force Push gesture
-EFFECT(monosfx);    // for Monophonically played sounds (iceblade, seismic charge etc...)
-EFFECT(swap);       // for standalone triggering EffectSequence<>
+EFFECT(tr);         // for EFFECT_TRANSITION_SOUND, use with User Effects.
 EFFECT(mute);       // Notification before muted ignition to avoid confusion.
+EFFECT(mzoom);      // for Spam Blast enter/exit
+
+template<class SPEC>
+struct BCVolumeMode : public SPEC::SteppedMode {
+  const int max_volume_ = VOLUME;
+  const int min_volume_ = VOLUME * 0.10;
+  float initial_volume_ = 0.0;
+  int initial_percentage_ = 0;
+  int percentage_ = 0;
+  bool announced_ = false;
+
+  int steps_per_revolution() override {
+    return 12;  // adjust for sensitivity
+  }
+
+  void mode_activate(bool onreturn) override {
+    PVLOG_NORMAL << "** Enter Volume Menu\n";
+    initial_volume_ = dynamic_mixer.get_volume();
+    initial_percentage_ = round((initial_volume_ / max_volume_) * 10) * 10;
+   SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    mode::getSL<SPEC>()->SayEnterVolumeMenu();
+    SPEC::SteppedMode::mode_activate(onreturn);
+  }
+
+  void mode_deactivate() override {
+    if (percentage_ <= 10) {
+      mode::getSL<SPEC>()->SayMinimumVolume();
+    } else if (percentage_ >=100) {
+      mode::getSL<SPEC>()->SayMaximumVolume();
+    } else {
+      mode::getSL<SPEC>()->SayWhole(percentage_);
+      mode::getSL<SPEC>()->SayPercent();
+    }
+    mode::getSL<SPEC>()->SayVolumeMenuEnd();
+    SPEC::SteppedMode::mode_deactivate();
+  }
+
+  void next() override {
+    int current_volume_ = dynamic_mixer.get_volume();
+    if (current_volume_ < max_volume_) {
+      current_volume_ += max_volume_ * 0.10;
+      if (current_volume_ > max_volume_) current_volume_ = max_volume_;
+      dynamic_mixer.set_volume(current_volume_);
+      mode::getSL<SPEC>()->SayVolumeUp();
+      announced_ = false;
+    } else if (!announced_) {
+      QuickMaxVolume();
+      announced_ = true;
+    }
+  }
+
+  void prev() override {
+    int current_volume_ = dynamic_mixer.get_volume();
+    if (current_volume_ > min_volume_) {
+      current_volume_ -= max_volume_ * 0.10;
+      if (current_volume_ < min_volume_) current_volume_ = min_volume_;
+      dynamic_mixer.set_volume(current_volume_);
+      mode::getSL<SPEC>()->SayVolumeDown();
```

Comment:
announced_ is to prevent it from repeating "Minimum/Maximum Volume" over and over. It says it once and then subsequent adjustments in the same direction say nothing until an opposite adjustment happens.
would it be better to name the variable like range_end_reached_ and/or add a comment explaining?

Reply:
But you don't need it if the code is changed so that it only says "maximum volume" if the volume 
1. was not at minimum/maximum
2. after incrementing/decrement, it reach minimum/maximum

Reply:
Yes ok.
I also moved the actual announce part out of deactivate into its own announce_volume() so it can be called both entering and exiting.
Woking well.